### PR TITLE
Adaptation text module and losses

### PR DESCRIPTION
**Pull request by leopoldmt**
Adaptation of the TextModule to make it more robust to small errors of predictions of the Global Workspace and be able to use MSE loss instead of token loss
Adaptation of unimodal losses to scale them if need to use token loss of text module

### CODE REVIEW

**Comment by RolandBERTINJOHANNET** [comment]
I don't understand a part of this commit -- I see the MSE Loss, but how is this part below done ?
> "make it more robust to small errors of predictions of the Global Workspace"

also I don't understand this sentence ?
> "Adaptation of unimodal losses to scale them if need to use token loss of text module"

**Comment by leopoldmt** [comment]
The first point is implemented in lines 407 to 410 in modules/domains/text.py

```
z = reparameterize(mean, logvar)
noise = torch.randn_like(z) * 0.5
z_text = z + noise
loss, acc = self.text_token_loss(z_text, x)
```

The second point is done with the code bellow in each module. It was originally done to scale if vectors were too big compared to others (especially the sequence of predicted tokens for each elements in the batch in the case of the original text module)

`LossOutput(F.mse_loss(pred, target, reduction="sum") / pred.numel())`

**Comment by leopoldmt** [comment]
Coming back to a normal MSE with mean reduction for more clarity

File: shimmer_ssd/modules/domains/text.py [hunk 1]
```diff
@@ -373,16 +403,35 @@ def text_token_loss(
     def generic_step(
         self, x: Mapping[str, torch.Tensor], mode: str = "train"
     ) -> torch.Tensor:
-        z = self.encode(x)
-        loss, acc = self.text_token_loss(z, x)
+        mean, logvar = self.encode_dist(x)
+        z = reparameterize(mean, logvar)
+        noise = torch.randn_like(z) * 0.5
+        z_text = z + noise
+        loss, acc = self.text_token_loss(z_text, x)
```
[the remark this answers is not in the document]
Comment (leopoldmt) [reply shimmer_ssd/modules/domains/text.py:408]: Indeed but it's only to pre-train the TextModel to be robust to noise. It's not used at inference time
